### PR TITLE
feat(ui): Improve aria attributes of menu items

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This makes menu items a little bit better

### CODE REVIEW

File: static/app/components/dropdownMenuItem.tsx [hunk 1]
```diff
@@ -199,6 +199,7 @@ const BaseDropdownMenuItem: React.ForwardRefRenderFunction<HTMLLIElement, Props>
 
   return (
     <MenuListItem
+      aria-haspopup={isSubmenuTrigger}
```
Comment (evanpurkhiser) [inline static/app/components/dropdownMenuItem.tsx:202]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup

Used to indicate that this `menuitem` is going to open a submenu

File: static/app/components/forms/controls/selectOption.tsx [hunk 2]
```diff
@@ -43,6 +43,8 @@ function SelectOption(props: Props) {
         <MenuListItem
           {...itemProps}
           {...innerProps}
+          role={isMultiple ? 'menuitemcheckbox' : 'menuitemradio'}
```
Comment (evanpurkhiser) [inline static/app/components/forms/controls/selectOption.tsx:46]: For selects (including compactSelect) they are checkboxes or radios.

File: static/app/components/forms/controls/selectOption.tsx [hunk 3]
```diff
@@ -43,6 +43,8 @@ function SelectOption(props: Props) {
...
+          aria-checked={isSelected}
```
Comment (evanpurkhiser) [inline static/app/components/forms/controls/selectOption.tsx:47]: `menuitemcheckbox` and `menuitemradio` require this attribute, which will be helpful in tests to understand if the item has been checked or not

File: static/app/components/menuListItem.tsx [hunk 4]
```diff
@@ -105,8 +106,19 @@ function BaseMenuListItem({
   forwardRef,
   ...props
 }: Props) {
+  const labelId = useMemo(() => domId('menuitem-label-'), []);
+  const detailId = useMemo(() => domId('menuitem-details-'), []);
+
   return (
-    <MenuItemWrap as={as} ref={forwardRef} {...props}>
+    <MenuItemWrap
+      role="menuitem"
```
Comment (evanpurkhiser) [inline static/app/components/menuListItem.tsx:114]: This will be overriden in `SelectOption`, but this can be considered the default role

**Review by ryan953 — APPROVED** [review]
:shipit:

**Comment by ryan953** [comment]
Would be nice if the storybook aria plugin was pointing these out, but i see things all passing in there: https://storybook.sentry.dev/?path=/story/components-buttons-dropdowns-dropdown-link--anchor-left-default

**Comment by evanpurkhiser** [comment]
going to try and update tests for this actually

File: static/app/components/dropdownMenuControl.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+          },
+        ]}
+        triggerLabel="This is a Menu"
+      />
+    );
+
+    // Open the mneu
+    userEvent.click(screen.getByRole('button', {name: 'This is a Menu'}));
+
+    // The mneu is open
```
Comment (priscilawebdev) [inline static/app/components/dropdownMenuControl.spec.tsx:31]:
```suggestion
    // The menu is open
```

File: static/app/components/dropdownMenuItem.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/components/dropdownMenuItem.tsx:202]: Nice!

**Pull request merged**